### DISCUSSION TITLE
Why the particle velocities are always biased in a scene?

### DISCUSSION THREAD

**Discussion by nikitanodar** [TheCodez/dynamic-occupancy-grid-map#98] [Q&A] [answered]
Hello. I have an issue that I struggle to understand. I am using a bird eye view images as an input, like this:

![0000007](https://github.com/TheCodez/dynamic-occupancy-grid-map/assets/132362566/239b2a3d-09ae-4ec5-8f16-13008fb4b705)

Here the white rectangle is an object - it is moved from right to left and this is the last frame in the sequence. I read that image and assign occupancy mass and free mass accordingly. In the end I have the following image:

![outputs_step_7](https://github.com/TheCodez/dynamic-occupancy-grid-map/assets/132362566/008b9c4d-176e-4714-9cf6-88bb546a8147)

Though it looks good, it's not perfect - I hope you can see the yellow-green gradient. And this happens for all the objects, the particle velocities are not consistent. But where does this inconsistency comes from?

Update: I looked closely to the original demo results:

![outputs_step_14](https://github.com/TheCodez/dynamic-occupancy-grid-map/assets/132362566/9142db2a-0951-45fd-8f63-6b7efa6cac8c)

The leftmost object obviously has the same problem - it has a smooth gradient from red to magenta. The same happens to the green object if you inspect pixel values.

**Comment by cbachhuber** [comment, marked as answer]
Great observation! I think this is a fundamental limitation of DOGMs: for example in [Figure 7.5 on page 119 of Nuss' Dissertation](https://d-nb.info/1133544290/34#page=119), we observe a similar gradient. This is because in larger objects, particles have a lot of space to freely move around in without dying. This creates spurious motion, often also observed in long, static guardrails next to roads.

I'm not aware of algorithmic remedies to this, but I'm also not up to date with the latest research.
